lib/node_modules/@stdlib/blas/ext/base/sindex-of/test/test.ndarray.js

```diff
@@ -0,0 +1,102 @@
+* you may not use this file except in compliance with the License.
+* You may obtain a copy of the License at
+*
+*    http://www.apache.org/licenses/LICENSE-2.0
+*
+* Unless required by applicable law or agreed to in writing, software
+* distributed under the License is distributed on an "AS IS" BASIS,
+* WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+* See the License for the specific language governing permissions and
+* limitations under the License.
+*/
+
+'use strict';
+
+// MODULES //
+
+var tape = require( 'tape' );
+var Float32Array = require( '@stdlib/array/float32' );
+var sindexOf = require( './../lib/ndarray.js' );
+
+
+// TESTS //
+
+tape( 'main export is a function', function test( t ) {
+	t.ok( true, __filename );
+	t.strictEqual( typeof sindexOf, 'function', 'main export is a function' );
+	t.end();
+});
+
+tape( 'the function returns the first index of an element which equals a provided search element', function test( t ) {
+	var actual;
+	var x;
+
+	x = new Float32Array( [ 1.0, 1.0, 2.0, 2.0, 3.0, 3.0 ] );
+
+	// Nonnegative stride...
+	actual = sindexOf( x.length, 1.0, x, 1, 0 );
+	t.strictEqual( actual, 0, 'returns expected value' );
+
+	actual = sindexOf( x.length, 2.0, x, 1, 1 );
+	t.strictEqual( actual, 2, 'returns expected value' );
+
+	actual = sindexOf( x.length, 3.0, x, 1, 2 );
+	t.strictEqual( actual, 4, 'returns expected value' );
+
+	// Negative stride...
+	actual = sindexOf( x.length, 1.0, x, -1, x.length-1 );
+	t.strictEqual( actual, 1, 'returns expected value' );
+
+	actual = sindexOf( x.length, 2.0, x, -2, x.length-1 );
+	t.strictEqual( actual, 3, 'returns expected value' );
+
+	actual = sindexOf( x.length, 1.0, x, -2, x.length-2 );
+	t.strictEqual( actual, 0, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided an empty array', function test( t ) {
+	var actual;
+
+	actual = sindexOf( 1, 0.0, new Float32Array( [] ), 0, 0 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided `N` parameter is less than or equal to zero', function test( t ) {
+	var actual;
+
+	actual = sindexOf( 0, 2.0, new Float32Array( [ 1.0, 2.0, 3.0 ] ), 1, 0 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	actual = sindexOf( -1, 2.0, new Float32Array( [ 1.0, 2.0, 3.0 ] ), 1, 0 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided a search element equal to `NaN`', function test( t ) {
+	var actual;
+
+	actual = sindexOf( 1, NaN, new Float32Array( [ NaN ] ), 1, 0 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	t.end();
+});
+
+tape( 'the function returns `-1` if provided an offset which exceeds the maximum array index', function test( t ) {
+	var actual;
+
+	actual = sindexOf( 3, 2.0, new Float32Array( [ 1.0, 2.0, 3.0 ] ), 1, 20 );
+	t.strictEqual( actual, -1, 'returns expected value' );
+
+	t.end();
+});
```

Comment:
This test should be removed. In strided array implementations, we don't perform buffer bounds checking.

Reply:
Again, it would be useful to refer to https://github.com/stdlib-js/stdlib/blob/develop/lib/node_modules/%40stdlib/blas/base/isamax/lib/ndarray.js.